docs/.vitepress/theme/custom.css

```
@@ -173,6 +173,24 @@
   }
 }
 
+/* Mermaid diagrams: fix text clipping (mermaid 11.15+ bugs #7354 / #7759).
+   - #7759: themeCSS lowercases `foreignObject` → `foreignobject`, but XML
+     CSS matching is case-sensitive, so we target both spellings.
+   - #7354: nodes don't expand vertically for long text; overflow:visible
+     + larger wrappingWidth (set in config.mts) mitigate this.
+   - VitePress body line-height (1.7) inflates rendered text vs measured
+     height, so we reset it inside mermaid containers. */
+.mermaid foreignObject,
+.mermaid foreignobject {
```

Stylelint (17.14.0)

[error] Unknown type selector "foreignobject" (selector-type-no-unknown)

(selector-type-no-unknown)